homeassistant/components/govee_light_api/light.py

```diff
@@ -0,0 +1,182 @@
+
+from govee_local_api import GoveeDevice
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ATTR_COLOR_TEMP_KELVIN,
+    ATTR_RGB_COLOR,
+    ColorMode,
+    LightEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .capability import (
+    GOVEE_COORDINATORS_MAPPER,
+    GOVEE_DEVICE_CAPABILITIES,
+    GoveeLightCapabilities,
+)
+from .const import DOMAIN, MANUFACTURER
+from .coordinator import GoveeLocalApiCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Govee light setup."""
+
+    coordinator: GoveeLocalApiCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    await coordinator.async_config_entry_first_refresh()
+
+    try:
+        async with asyncio.timeout(delay=5):
+            while not coordinator.devices:
+                await asyncio.sleep(delay=1)
+    except asyncio.TimeoutError:
+        pass
+
+    async_add_entities(
+        GoveeLight(coordinator, device) for device in coordinator.devices
+    )
+
+    def discovery_callback(device: GoveeDevice, is_new: bool) -> bool:
+        if is_new:
+            async_add_entities([GoveeLight(coordinator, device)])
+        return True
+
+    await coordinator.set_discovery_callback(discovery_callback)
+
+
+class GoveeLight(CoordinatorEntity, LightEntity):
+    """Govee Light."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: GoveeLocalApiCoordinator,
+        device: GoveeDevice,
+    ) -> None:
+        """Govee Light constructor."""
+
+        super().__init__(coordinator)
+        self._device = device
+        self._device.set_update_callback(self._update_callback)
+
+        capabilities: set[GoveeLightCapabilities] = GOVEE_DEVICE_CAPABILITIES.get(
+            device.sku, set()
+        )
+
+        self._attr_unique_id: str = device.fingerprint
+
+        color_modes = set()
+        if GoveeLightCapabilities.COLOR_RGB in capabilities:
+            color_modes.add(ColorMode.RGB)
+        if GoveeLightCapabilities.COLOR_KELVIN_TEMPERATURE in capabilities:
+            color_modes.add(ColorMode.COLOR_TEMP)
+            self._attr_max_color_temp_kelvin = 9000
+            self._attr_min_color_temp_kelvin = 2000
+        if GoveeLightCapabilities.BRIGHTNESS in capabilities:
+            color_modes.add(ColorMode.BRIGHTNESS)
+
+        if len(color_modes) == 0:
+            color_modes.add(ColorMode.ONOFF)
+
+        self._attr_supported_color_modes: set[ColorMode] | set[str] = color_modes
+
+    @property
+    def name(self) -> str:
+        """Name of the entity."""
+        return self._device.sku
```

Comment:
device_info can be set in the constructor with `_attr_device_info`. Then you can set `_attr_name = None` and the entity will follow the device name